lib/languages/sass.js

```diff
@@ -0,0 +1,180 @@
+var DocsParser = require("../docsparser");
+var xregexp = require('../xregexp').XRegExp;
+
+function SassParser(settings) {
+    // this.setup_settings();
+    // call parent constructor
+    DocsParser.call(this, settings);
+}
+
+SassParser.prototype = Object.create(DocsParser.prototype);
+
+SassParser.prototype.setup_settings = function() {
+    var identifier = '[a-zA-Z_-][a-zA-Z_0-9-]*';
+    this.settings = {
+        'commentType': 'single',
+        'curlyTypes': true,
+        'typeInfo': true,
+        'typeTag': 'type',
+        'prefix': '///',
+        'varIdentifier': '\\$' + identifier,
+        'fnIdentifier':  identifier,
+        'fnOpener': '@(?:mixin|function)\\s+' + identifier + '\\s*[\\(|\\{]',
+        'bool': 'Boolean'
+    };
+};
+
+SassParser.prototype.parse_function = function(line) {
+    var r = '^\\s*@(?P<fnType>(mixin|function))\\s+' +
+            '(?P<name>' + this.settings.fnIdentifier + ')' +
+            '\\s*(?:\\(\\s*(?P<args>.*?)\\)|{)';
+    var regex = xregexp(r);
+
+    var matches = xregexp.exec(line, regex);
+    if(matches === null)
+        return null;
+
+    return [
+        matches.name,
+        (matches.args ? matches.args.trim() : null),
+        (matches.fnType == "mixin" ? null : false)
+    ];
+};
+
+SassParser.prototype.parse_var = function(line) {
+    //   $foo: blah;
+    var r = '(?P<name>' + this.settings.varIdentifier + ')' +
+            '\\s*:\\s*(?P<val>.*?)(?:;|$)';
+    var regex = xregexp(r);
+
+    var matches = xregexp.exec(line, regex);
+
+    if(matches === null) {
+        return null;
+    }
+    // variable name, variable value
+    return [matches.name, matches.val.trim()];
+};
+
+SassParser.prototype.parse_arg = function(arg) {
+
+    var regex = xregexp(
+        '(?P<name>' + this.settings.varIdentifier + ')(\\s*:\\s*(?P<value>.*))?'
+    );
+
+    return xregexp.exec(arg, regex);
+};
+
+SassParser.prototype.get_arg_type = function(arg) {
+
+    var matches = this.parse_arg(arg);
+
+    if(matches && matches.value) {
+        return this.guess_type_from_value(matches.value);
+    }
+
+    return null;
+};
+
+SassParser.prototype.get_arg_name = function(arg) {
+
+    var matches = this.parse_arg(arg);
+
+    if (matches && matches.value) {
+      return util.format('%s [%s]', matches.name, matches.value);
+    }
+
+    return matches.name;
+};
+
+SassParser.prototype.guess_type_from_value = function(val) {
+    var lower_primitives = this.editor_settings.lower_case_primitives || false;
+    var short_primitives = this.editor_settings.short_primitives || false;
+    var var_type;
+    var capitalize = function(str) {
+        return str.charAt(0).toUpperCase() + str.slice(1);
+    };
+
+    // It doesn't detect string without quotes, because it could be a color name
+    // It doesn't detect empty lists or maps, because they have the same syntax
+
+    if(this.is_map(val)) {
+        var_type = 'map';
+        return (lower_primitives ? var_type : capitalize(var_type));
```

Comment:
As far as I know the `lower_primitives` setting was only added for JS, because JSDoc didn't clearly define this (jsdoc3/jsdoc#1046) but some tools like the Google Closure Compiler expect lower case primitives to differentiate between wrapper classes. For scss, as far as I can tell, this is not necessary and should be ignored.